src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp

```diff
@@ -85,6 +102,117 @@ namespace Tests
         ASSERT_EQ(MMI_OK, mm->MpiSet(componentName1.c_str(), "", payload, ARRAY_SIZE(payload)));
     }
 
+    TEST_F(ModuleManagerTests, MpiSetDesiredSingleComponent)
+    {
+        constexpr const char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    }
+                }
+            ])"""";
+
+        ASSERT_EQ(MPI_OK, mm->MpiSetDesired(clientName.c_str(), payload, ARRAY_SIZE(payload)));
+    }
+
+    TEST_F(ModuleManagerTests, MpiSetDesiredMultipleComponents)
+    {
+        constexpr const char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    },
+                    "TestComponent2": {
+                        "testParameter": "testValue"
+                    }
+                }
```

Comment:
This is incorrect multiple componenst case. 

Multiple  components could be CommandRunner and Firewall, for example.

Use real looking MIM components and objects and settings, please.

And add complex objects to the simple ones